From 18c557fbdd19b3ee872c14831d6117e457933ae6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 4 Dec 2018 14:59:14 +0100
Subject: [PATCH 30/55] gettext: Only use domain for our own translations

imports.gettext.textdomain() changes the domain for any translations
in the gnome-shell process, which is clearly not what an extension
should do. Instead, just make sure that the _() macro we define uses
the expected domain without affecting any code outside the extension.

https://gitlab.gnome.org/World/ShellExtensions/desktop-icons/issues/66
---
 desktopGrid.js | 4 +---
 fileItem.js    | 4 +---
 prefs.js       | 4 +---
 3 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/desktopGrid.js b/desktopGrid.js
index 1e7f278..ffc42fc 100644
--- a/desktopGrid.js
+++ b/desktopGrid.js
@@ -43,9 +43,7 @@ const Util = imports.misc.util;
 
 const Clipboard = St.Clipboard.get_default();
 const CLIPBOARD_TYPE = St.ClipboardType.CLIPBOARD;
-const Gettext = imports.gettext;
-
-Gettext.textdomain('desktop-icons');
+const Gettext = imports.gettext.domain('desktop-icons');
 
 const _ = Gettext.gettext;
 
diff --git a/fileItem.js b/fileItem.js
index c45ffb3..bbe2587 100644
--- a/fileItem.js
+++ b/fileItem.js
@@ -42,9 +42,7 @@ const Prefs = Me.imports.prefs;
 const DBusUtils = Me.imports.dbusUtils;
 const DesktopIconsUtil = Me.imports.desktopIconsUtil;
 
-const Gettext = imports.gettext;
-
-Gettext.textdomain('desktop-icons');
+const Gettext = imports.gettext.domain('desktop-icons');
 
 const _ = Gettext.gettext;
 
diff --git a/prefs.js b/prefs.js
index 5408728..4fb1724 100644
--- a/prefs.js
+++ b/prefs.js
@@ -26,9 +26,7 @@ const Gettext = imports.gettext;
 
 const Config = imports.misc.config;
 
-Gettext.textdomain('desktop-icons');
-
-var _ = Gettext.gettext;
+var _ = Gettext.domain('desktop-icons').gettext;
 
 const SCHEMA_NAUTILUS = 'org.gnome.nautilus.preferences';
 const SCHEMA_GTK = 'org.gtk.Settings.FileChooser';
-- 
2.20.0

